uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -909,39 +912,37 @@ public static function findOverlappingLeaveRequests($leaveRequestParams, $leaveR
     }
 
     $dateTimeSelector = $isCalculationUnitInHoursOrTOIL
-      ? "lr.from_date < '{$toDate}' AND lr.to_date > '{$fromDate}'"
-      : 'lrd.date BETWEEN %3 AND %4';
+      ? "lr.from_date < %7 AND lr.to_date > %6"
+      : "lrd.date BETWEEN %4 AND %5";
 
-    $absenceTypesIDs = array_merge($absenceTypesIDsWithSameUnit, $absenceTypesIDsWithDifferentUnit);
-    $absenceTypesSelector = $isCalculationUnitInHoursOrTOIL
-      ? (!$isTOIL
-        ? 'AND lr.type_id IN (' . implode(',', $absenceTypesIDsWithSameUnit) . ')'
-        : '')
-      : 'AND lr.type_id IN (' . implode(',', $absenceTypesIDs) . ')';
+    if ($isCalculationUnitInHours) {
+      $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDsWithSameUnit) . ')';
 
-    $considerRequestsWithDifferentUnit = count($absenceTypesIDsWithDifferentUnit) && !$isTOIL
-      ? "OR ((lrd.date BETWEEN %3 AND %4) AND lr.type_id IN (" .
-        implode(',', $absenceTypesIDsWithDifferentUnit) . "))"
-      : "";
+      if (count($absenceTypesIDsWithDifferentUnit)) {
+        $absenceTypesSelector .=
```

Comment:
The name for this variable seems wrong somehow, its selects absence type and dates, maybe a better name?

Reply:
You, that's why I had `$considerRequestsInDifferentUnit`. I put it back, I think now it look neat enough.